### ISSUE TITLE
weird kernel dispatch for gradient of 3-interpolate

### ISSUE THREAD

**Issue by AluminiumUnibody** [facebookresearch/votenet#36]
It appears that gradient kernel was never actually used cuz here it's dispatching the same kernel as in forward:

https://github.com/facebookresearch/votenet/blob/3992eeac3e4cdd21d5500bcc7642b559292f540c/pointnet2/_ext_src/src/interpolate.cpp#L47-L75
https://github.com/facebookresearch/votenet/blob/3992eeac3e4cdd21d5500bcc7642b559292f540c/pointnet2/_ext_src/src/interpolate.cpp#L76-L104

Specifically I think this line:

https://github.com/facebookresearch/votenet/blob/3992eeac3e4cdd21d5500bcc7642b559292f540c/pointnet2/_ext_src/src/interpolate.cpp#L95

should have been ```three_interpolate_grad_kernel_wrapper``` instead? Is this a mistake or intentionally done as such?

**Comment by charlesq34** [comment]
@AluminiumUnibody 

Good catch!

I think it seems a bug/typo. It happens that the shape matches so there is no explicit error.

Unfortunately I don't have time right now to examine into this issue.
It would be very helpful if you have time to take a deeper look! e.g. use the `three_interpolate_grad_kernel_wrapper` and try to verify the gradients or directly re-run the training experiment.

**Comment by AluminiumUnibody** [comment]
@charlesq34 Okay thanks. It's quite interesting that it didn't seem to have affected the training much.

**Comment by charlesq34** [comment]
I looked into the issue and found that it is a bug. It has now been fixed.

I've added `pointnet2/pointnet2_test.py' to gradient check the fixed `three_interpolate` ops.
Re-training with the fixed ops resulted in a slightly improved model. With the eval code in `train.py`, we can get around 58.6 mAP@0.5 for SUN RGB-D (although evaluation with `eval.py` seems not show such big improvement).

Thank you @AluminiumUnibody for reporting this issue!
